api/src/main/java/org/apache/iceberg/rest/CreateNamespaceRequest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.io.Serializable;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+/**
+ * Represents a REST request to create a namespace / database.
+ */
+public class CreateNamespaceRequest implements Serializable {
+
+  // TODO - Use protected so users can extend this for their own impls. Or an interface.
+  //        Currently anything but private causes an error.
+  private String namespaceName;
+  private Map<String, String> properties;
+
+  private CreateNamespaceRequest() {
+
+  }
+
+  private CreateNamespaceRequest(String namespaceName, Map<String, String> properties) {
```

Comment:
The `Catalog` API allows a namespace that is any number of levels. Should this be `Namespace`, `List<String>`, or `String[]`?

Reply:
It makes sense to use `Namespace` to me, but in the JSON, I'd prefer it to be a `List<String>` or `String[]` so users can specify as many levels as they'd like.

Will update.

Reply:
I will come back and update these things after we discuss the proposed specification  👍